micro-rdk/src/common/camera.rs

```diff
@@ -52,11 +71,67 @@ impl Default for FakeCamera {
     }
 }
 
+impl Camera for FakeCamera {
+    fn get_image(&mut self, mut buffer: BytesMut) -> Result<BytesMut, CameraError> {
+        #[cfg(feature = "camera")]
+        let image = FAKE_JPEG;
+        #[cfg(not(feature = "camera"))]
+        let image = BytesMut::new();
+        let msg = camera::v1::GetImageResponse {
+            mime_type: "image/jpeg".to_string(),
+            image: image.into(),
+        };
+        msg.encode(&mut buffer).unwrap();
+        Ok(buffer)
+    }
+    fn get_images(&mut self, _buffer: BytesMut) -> Result<BytesMut, CameraError> {
+        unimplemented!();
+    }
+    fn get_point_cloud(&mut self, _buffer: BytesMut) -> Result<BytesMut, CameraError> {
+        unimplemented!();
+    }
+    fn get_properties(&mut self, _buffer: BytesMut) -> Result<BytesMut, CameraError> {
+        unimplemented!();
+    }
+    fn render_frame(&mut self, mut buffer: BytesMut) -> Result<BytesMut, CameraError> {
+        #[cfg(feature = "camera")]
+        let msg = google::api::HttpBody {
+            content_type: "image/jpeg".to_string(),
+            data: FAKE_JPEG.into(),
+            ..Default::default()
```

Comment:
do we make any use of HTTP extensions?

Reply:
not in [rdk camera server](https://github.com/viamrobotics/rdk/blob/04037e3ddc9abd2740b1e3d5f3e8152e04e3b121/components/camera/server.go#L194) at least